### PR TITLE
Add generic asset rpc call for querying asset meta data

### PR DESCRIPTION
**Pull request by aliXsed**
This adds Generic Asset to plug-blockchain runtime and testnet, as wellG

### CODE REVIEW

File: bin/node/cli/src/chain_spec.rs [hunk 1]
```diff
@@ -289,6 +288,20 @@ pub fn testnet_genesis(
...
 		pallet_membership_Instance1: Some(Default::default()),
 		pallet_treasury: Some(Default::default()),
+		pallet_generic_asset: Some(GenericAssetConfig {
+			assets: vec![STAKING_ASSET_ID, SPENDING_ASSET_ID],
+			// Grant root key full permissions (mint,burn,update) on the following assets
+			permissions: vec![
+				(STAKING_ASSET_ID, root_key.clone()),
+				(SPENDING_ASSET_ID, root_key.clone()),
+			],
+			initial_balance: 10u128.pow(18 + 9), // 1 billion token with 18 decimals
```
Comment (aliXsed) [inline bin/node/cli/src/chain_spec.rs:305]: 18 or 6?

File: frame/generic-asset/rpc/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+use sp_core::Bytes;
+use pallet_generic_asset::AssetInfo;
+pub use pallet_generic_asset_rpc_runtime_api::AssetMetaApi;
+pub use self::gen_client::Client as GenericAssetClient;
+
+#[rpc]
+pub trait GenericAssetApi<BlockHash, ResponseType>
+{
+	/// Get all assets data paired with their ids.
+	#[rpc(name = "generic_asset_meta")]
```
Comment (jordy25519) [inline frame/generic-asset/rpc/src/lib.rs]: I think the convention is camelCase: `genericAsset_assetMeta`, might be nice as `genericAsset_registeredAssets`

File: frame/generic-asset/rpc/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	for GenericAsset<C, (Block, AssetId)>
+where
+	Block: BlockT,
+	C: Send + Sync + 'static + ProvideRuntimeApi<Block> + HeaderBackend<Block>,
+	C::Api: AssetMetaApi<Block, AssetId>,
+	AssetId: Decode + Encode + Send + Sync + 'static,
+{
+	fn asset_meta(
+		&self,
+		_encoded_xt: Bytes,
```
Comment (jordy25519) [inline frame/generic-asset/rpc/src/lib.rs]: can remove this arg

**Review by jordy25519 — APPROVED** [review]
changes lgtm!

File: frame/generic-asset/src/tests.rs [hunk 4]
```diff
@@ -1197,6 +1197,18 @@ fn total_issuance_should_update_after_negative_imbalance_dropped() {
...
 }
 
+#[test]
+fn query_pre_existing_asset_info() {
+	ExtBuilder::default()
+		.free_balance((STAKING_ASSET_ID, ALICE, INITIAL_BALANCE))
+		.build()
+		.execute_with(|| {
+			type AssetId = <Test as Trait>::AssetId;
+			let registered_assets: Vec<(AssetId, AssetInfo)> = <AssetMeta<Test> as IterableStorageMap<AssetId,AssetInfo>>::iter().collect();
```
Comment (jordy25519) [inline frame/generic-asset/src/tests.rs]: This could go into a function and be called by the RPC impl so then the unit test is testing the same code

Comment (aliXsed) [reply frame/generic-asset/src/tests.rs]: If you are okay, let's leave this to future improvements. I just don't want to restart the CircleCi build.

Comment (aliXsed) [reply frame/generic-asset/src/tests.rs]: Actually, as I need to fix the test build, I'm going to address this one too.

**Pull request merged**